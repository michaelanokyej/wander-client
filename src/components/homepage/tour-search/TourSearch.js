import React from "react";

class TourSearch extends React.Component {
  render() {
    return (
      <section>
        <header>
          <h3>Find a tour</h3>
          <form class="tour-search-form">
            <div>
              <h4>Location</h4>
              <div>
                <label for="city">City</label>
                <input placeholder='Washington' type="text" name='city' id='city' />
              </div>
              <div>
                <label for="state">State</label>
                <input
                  placeholder="DC"
                  type="text"
                  name="state"
                  id="state"
                  maxlength="2"
                />
              </div>
              <div>
                <h4>Dates</h4>
                <div>
                  <label for="begin">Tour start</label>
                  <input type="date" name='begin' id='begin' placeholder='MM/DD/YYYY'/>
                </div>
                <div>
                  <label for="end">Tour End</label>
                  <input type="date" name='end' id='end' placeholder='MM/DD/YYYY'/>
                </div>
                <div>
                  {/* <input type="radio" name="group" value="group"/>
                  <input type="radio" name="private" value="private"/>  */}
                  <label for="group">
                  <input type="radio" name="group" value="group"/>
                  Group Tour
                  </label> 
                  <label for="private">
                  <input type="radio" name="private" value="private"/>
                  Private Tour
                  </label>
                </div>
              </div>
            </div>
            <button type='submit'>Find A Tour</button>
          </form>
        </header>
      </section>
    );
  }
}

export default TourSearch;
