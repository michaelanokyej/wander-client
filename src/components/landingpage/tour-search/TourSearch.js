import React from "react";

class TourSearch extends React.Component {
  render() {
    return (
      <section>
        <header>
          <h3>Find a tour</h3>
          <form className="tour-search-form">
            <div>
              <h4>Location</h4>
              <div>
                <label htmlFor="city">City</label>
                <input placeholder='Washington' type="text" name='city' id='city' />
              </div>
              <div>
                <label htmlFor="state">State</label>
                <input
                  placeholder="DC"
                  type="text"
                  name="state"
                  id="state"
                  maxLength="2"
                />
              </div>
              <div>
                <h4>Dates</h4>
                <div>
                  <label htmlFor="begin">Tour start</label>
                  <input type="date" name='begin' id='begin' placeholder='MM/DD/YYYY'/>
                </div>
                <div>
                  <label htmlFor="end">Tour End</label>
                  <input type="date" name='end' id='end' placeholder='MM/DD/YYYY'/>
                </div>
                <div>
                  {/* <input type="radio" name="group" value="group"/>
                  <input type="radio" name="private" value="private"/>  */}
                  <label htmlFor="group">
                  <input type="radio" name="group" value="group"/>
                  Group Tour
                  </label> 
                  <label htmlFor="private">
                  <input type="radio" name="private" value="private"/>
                  Private Tour
                  </label>
                </div>
              </div>
            </div>
            <button type='submit'>Find A Tour</button>
          </form>
        </header>
      </section>
    );
  }
}

export default TourSearch;
