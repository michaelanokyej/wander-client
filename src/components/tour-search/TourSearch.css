.tourSearchSection {
  width: 100%;
  height: 30vh;
  background: url(banner2.jpg);
  background-size: cover;
}

.tour-search-from {
  background: transparent;
}
