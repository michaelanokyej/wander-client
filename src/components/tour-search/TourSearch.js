import React from "react";
import "./TourSearch.css";
import toursAndUserContext from "../context/toursAndUserContext.js";
import { Redirect } from "react-router-dom";
import Swal from "sweetalert2/dist/sweetalert2.js";
import "sweetalert2/src/sweetalert2.scss";

class TourSearch extends React.Component {
  static contextType = toursAndUserContext;

  state = {
    city: "",
    state: "",
    routeToResultsPage: false
  };
  render() {
    const searchInfo = {
      city: this.state.city,
      state: this.state.state
    };
    return (
      <section className="tourSearchSection">
        {this.state.routeToResultsPage && <Redirect to="/search/tours" />}

        <form className="tour-search-from">
          <div className="div_form">
            <input
              required
              type="text"
              name="city"
              id="city"
              onChange={e => this.setState({ city: e.target.value })}
            />
            <label htmlFor="city" className="label-name">
              <span className="content-name">City</span>
            </label>
          </div>
          <div className="div_form">
            <input
              required
              type="text"
              name="state"
              id="state"
              maxLength="2"
              onChange={e => this.setState({ state: e.target.value })}
            />
            <label htmlFor="state" className="label-name">
              <span className="content-name">State</span>
            </label>
          </div>
          <button
            className="post-tour-button"
            type="submit"
            onClick={e => {
              e.preventDefault();
              if (
                this.state.city.length !== 0 &&
                this.state.state.length !== 0
              ) {
                this.context.handleTourSearch(searchInfo);
                this.setState({
                  routeToResultsPage: true
                });
              } else {
                Swal.fire("city and state are required to search");
              }
            }}
          >
            Search Tours
          </button>
        </form>
      </section>
    );
  }
}

export default TourSearch;
