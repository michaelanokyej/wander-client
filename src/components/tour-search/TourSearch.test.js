import React from "react";
import TourSearch from "./TourSearch";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe(`TourSearch component`, () => {
  it(`renders the TourSearch component`, () => {
    const wrapper = shallow(<TourSearch />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
