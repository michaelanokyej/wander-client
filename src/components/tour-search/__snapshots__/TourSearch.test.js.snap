// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TourSearch component renders the TourSearch component 1`] = `
<section
  className="tourSearchSection"
>
  <form
    className="tour-search-from"
  >
    <div
      className="div_form"
    >
      <input
        id="city"
        name="city"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <label
        className="label-name"
        htmlFor="city"
      >
        <span
          className="content-name"
        >
          City
        </span>
      </label>
    </div>
    <div
      className="div_form"
    >
      <input
        id="state"
        maxLength="2"
        name="state"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <label
        className="label-name"
        htmlFor="state"
      >
        <span
          className="content-name"
        >
          State
        </span>
      </label>
    </div>
    <button
      className="post-tour-button"
      onClick={[Function]}
      type="submit"
    >
      Search Tours
    </button>
  </form>
</section>
`;
